server/application/terminal.go

```diff
@@ -120,18 +136,21 @@ func (s *terminalHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	var findContainer bool
+	var foundContainerName string
 	for _, c := range pod.Spec.Containers {
 		if container == c.Name {
-			findContainer = true
+			foundContainerName = c.Name
 			break
 		}
 	}
-	if !findContainer {
+	if foundContainerName == "" {
 		http.Error(w, "Cannot find container", http.StatusBadRequest)
```

Comment:
For auditing it would be interesting to also have this logged (as a warning) with all info: cluster, namespace, pod name and container name.

Reply:
CodeQL might complain about logging the un-sanitized `container` input. But since we're validating the container name above, I think it would be safe to log (and to override the CodeQL warning).

Reply:
@leoluz - yes this is the issue we were trying to avoid - ```Log entries created from user input High
This log write receives unsanitized user input from here.```
It is unsafe to log verbatim whatever was posted to URL params which is why it is not logged. 
In theory it should rarely happen as most requests should come directly from the argocd application itself and only post namespace + pod + container as already found in argocd.  I agree it would be useful to be able to debug though.  If a user really wants to know why it can't be found wondering if there is wireshark/tcpdump some level of capturing the http request instead?

Reply:
@leoluz and @crenshaw-dev - totally make sense - we've already validated so it's OK to log these.  I have added the additional info to these logs.